processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java

```diff
@@ -0,0 +1,36 @@
+package com.redhat.service.smartevents.processor;
+
+import java.util.stream.Collectors;
+
+import com.networknt.schema.ValidationMessage;
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+import com.redhat.service.smartevents.infra.validations.ValidationResult;
+
+public abstract class AbstractGatewayValidator<T extends Gateway> implements GatewayBean {
+
+    private ProcessorCatalogService processorCatalogService;
+
+    public AbstractGatewayValidator() {
+    }
+
+    public AbstractGatewayValidator(ProcessorCatalogService processorCatalogService) {
+        this.processorCatalogService = processorCatalogService;
+    }
+
+    protected ValidationResult applyAdditionalValidations(T gateway) {
+        return ValidationResult.valid();
```

Comment:
If are there any additional validations to be applied by an action/source, the processor validator has to override this method